comment_scraper.py

```diff
@@ -1,87 +1,96 @@
-                            object.update({'name':asset_name})
-                        comment_list.append(my_comment_list)
+def flatten_dict(d):
+    # The get_comments API response is verbose and contains nested objects.
+    # Use this helper functon to flatten the dict holding the API response data
+    # and namespace the attributes.
 
-# Takes a root asset ID for a project and a developer token.
-# Returns a comment list with all assets.
+    def expand(key, val):
+            if isinstance(val, dict):
+                return [ (key + '.' + k, v) for k, v in flatten_dict(val).items() ]
+            else:
+                return [ (key, val) ]
+    
+    items = [ item for k, v in d.items() for item in expand(k, v)]
 
-def get_all_project_comments(root_asset_id, token):
-    comment_list = []
-    client = FrameioClient(token)
+    return dict(items)
 
-    all_comments(client, root_asset_id, comment_list)
 
-    return comment_list
+def write_comments_csv(c_list):
+    # Writes comments to comments.csv
+    # Any attributes you add to the headers list will automatically be written to the CSV
+    # The API returns many attributes so familiarize with the response data!
+    headers = ['text', 'name', 'inserted_at', 'timestamp', 'has_replies', 'parent_id', 'owner.name', 'owner.email', 'owner_id', 'owner.account_id']
 
-# Get all the comments on the project you choose by providing a root asset ID and a developer token.
+    # Flattening the comments dicts is not at all necessary, but the namespacing
+    # makes the CSV headers much more readable.
+    flat_comments_list = []
+    for c in c_list:
+        flat_comments_list.append(flatten_dict(c))
 
-responses = get_all_project_comments(ROOT_ASSET_ID, TOKEN)
+    with open('comments.csv', 'w') as file:
+        f_csv = csv.DictWriter(file, headers, extrasaction='ignore')
+        f_csv.writeheader()
+        f_csv.writerows(flat_comments_list)
 
-# The response list comes back as a list of lists.
-# Flatten out responses so that there's only one item in each part of the list
 
-flat_response_list = list(itertools.chain.from_iterable(responses))
+# Initialize the client library
+client = FrameioClient(TOKEN)
 
-# Now we can use list comprehension to grab what we want from each block in the list and make a new flat list.
-list_for_csv = [[o['text'], o['parent_id'], o['asset_id'], o['name'], o['owner_id'], o['owner']['email'], o['timestamp'], o['updated_at']] for o in flat_response_list]
+# Build the comments list
+comments = []
+comments_list = build_comments_list(client, ROOT_ASSET_ID, comments)
 
-# Let's write our new list out to a .csv file. We'll add a heading.
-with open("output.csv", 'w') as myfile:
-     wr = csv.writer(myfile, dialect='excel')
-     wr.writerow(['Comment', 'Parent ID', 'Asset ID', 'Asset Name', 'Owner ID', 'Email', 'Timestamp', 'Updated At'])
-     wr.writerows(list_for_csv)
+# Write the comments to comments.csv
+write_comments_csv(comments_list)
```

Comment:
one thing id be thinking about when building a script like this is decoupling the heavy lifting (fetching from the API and reformatting the results into a list of csv records) from how the script is invoked (e.g. on the command line vs. from a test vs. imported by another module) and what configuration is requested (the token in an environment variable, output sent to a file, etc). think about how you would test this from another python file without actually having to look at an output file on the file system. so a first step is to hide the actual calls to your stuff in:
```python
if __name__ == "__main__":
```
so it doesn't get executed if the file is simply imported instead of run. 
then i'd probably separate the functionality into two chunks: (1) setting up configuration, fetching tokens from the environment, specifying an output file path perhaps as an iostream so you can also direct to stdout, (2) calling the api, formatting the results, and sending them to the specified output stream. this would allow someone else (or you in the future, which is basically the same thing), to import this file, pass in an in-memory stream writer thing (i cant remember what these are called in python) and a token, and get back flattened formatted comments data to work with. no need to execute the script in a subprocess and then look for a specially named file to read into memory.